### DISCUSSION TITLE
Bug? EditTextWidget allows empty textreferences

### DISCUSSION THREAD

**Discussion by twMat** [TiddlyWiki/TiddlyWiki5#5621] [General]
This creates an unnamed field in foo. Maybe this shouldn't be allowed?

`<$edit-text tiddler="foo" field="" />`

In contrast, if I directly edit a tiddler and try to create a custom field by entering a field value but skip the field name, then this is not accepted.

It is also not possible to delete this unnamed field with the dustbin in the custom field area.

Further, `<$edit-text tiddler="" />` doesn't seem to create anything, but there is no alert or anything. Interestingly, the editor does show the `Type the text for the tiddler ''` message. Makes me wonder which tiddlers text it is looking at?

**Comment by kookma** [comment]
Confirmed! I believe this is a bug!

**Creating fields programmatically** has many issues and need to be checked thoroughly in the whole TW core!

Another bug

```
<$edit-text tiddler="LoLi" />
```

creates a filed with CamelCase as `LoLi` while field name in CamelCase or uppercase is not allowed in Tiddlywiki!
